pkg/miner/pool.go

```diff
@@ -0,0 +1,97 @@
+package miner
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-log"
+)
+
+// Persistence defines the expected interface for storing and loading generated
+// and not-yet-used parameters on persistent storage. Generating parameters is
+// usually a computationally expensive operation and generated parameters should
+// survive client restarts.
+type Persistence[T any] interface {
```

Comment:
Shouldn't we add some type constraints for `T`? I think we should make sure that this type can be persisted, i.e. implements `Marshal` and `Unmarshal` functions. WDYT?

Reply:
I was thinking about it but when we would assume how `Persistence` works internally. It is possible some implementation does not want/need to go to bytes to persist.